sma_set.go

```diff
@@ -0,0 +1,50 @@
+package movavg
+
+// SMASet is a set of related calculators of Simple Moving Average:
+// https://en.wikipedia.org/wiki/Moving_average#Simple_moving_average
+//
+// It is as precise, as float64 is: https://golang.org/pkg/builtin/#float64 (good performance, not-so-good precision).
+//
+// It is not thread-safe, if you need to access it from multiple goroutines - secure it with ThreadSafeSet(...).
+//
+// It is safe to use even without adding values (then Avg() will return zeroes),
+// or without filling entire window (then average of currently added values will be returned).
+//
+// To initialize calculator with current average value, simply Add(...) this value.
+type SMASet struct {
+	smas []*SMA
+}
+
+// NewSMASet constructs new Simple Moving Average calculator set.
+// Window arg must have 1 or more elements. For example, windows
+// of 1 minute, 5 minutes, 30 minutes, 1 hour for a given measurement.
+func NewSMASet(windows []int) *SMASet {
+	if windows == nil || len(windows) == 0 {
```

Comment:
I suggest this one to be removed (see comment above regarding move abstract `Set`/`NewSet`), but no need to check slice for `nil`, `len` would be more than enough.

Also, theoretically, no need to `panic` in case of zero windows... (not really sure about this, that's quite useful)